File: 219-contains-duplicate-ii/contains-duplicate-ii.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        dist = {}
        for i in range(len(nums)):
            if nums[i] in dist:
                if abs(dist[nums[i]]- i) <=k:
                    return True

            dist[nums[i]]= i
        return False

File: 219-contains-duplicate-ii/contains-duplicate-ii.java
import java.util.*;

class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> dist = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (dist.containsKey(nums[i])) {
                if (Math.abs(dist.get(nums[i]) - i) <= k) {
                    return true;
                }
            }
            dist.put(nums[i], i);
        }
        return false;
    }
}

